packages/engine/src/hash_types/state.rs

```diff
@@ -23,27 +23,33 @@ pub type DatasetMap = serde_json::Map<String, serde_json::Value>;
 
 // NOTE: This is used in conjunction with the custom deserializaer
 // PLEASE UPDATE THIS LIST WHEN YOU ADD ANOTHER BUILT IN FIELD
+/// Fields built into the engine.
+///
+/// Also see [`AgentStateField`].
 pub const BUILTIN_FIELDS: [&str; 12] = [
-    "agent_id",
-    "agent_name",
-    "messages",
-    "position",
-    "direction",
-    "velocity",
-    "shape",
-    "height",
-    "scale",
-    "color",
-    "rgb",
-    "hidden",
+    AgentStateField::AgentId.name(),
+    AgentStateField::AgentName.name(),
+    AgentStateField::Messages.name(),
+    AgentStateField::Position.name(),
+    AgentStateField::Direction.name(),
+    AgentStateField::Velocity.name(),
+    AgentStateField::Shape.name(),
+    AgentStateField::Height.name(),
+    AgentStateField::Scale.name(),
+    AgentStateField::Color.name(),
+    AgentStateField::RGB.name(),
+    AgentStateField::Hidden.name(),
 ];
 
-/// TODO: DOC, reword below to be a docstring
-// We also have context -- for an Agent, this is the world around it.
-// Context is where we find things like neighbors, properties, and any data not stored in an agent
-// itself. Our agent runner tries hard to only give agents the context they need to execute, nothing
-// more. Context is immutable, although an agent can send messages to other agents as part of its
-// return value.
+/// The context is the parts of the simulation that the agent is exposed to.
+///
+/// Another way of thinking about it is that an agent's context are the parts of the simulation that
+/// the agent "can see". It is an immutable object, which is passed to each behavior and has data
+/// not stored in an agent itself like neighboring information, message handling, and globals.
+///
+/// Agent runners tries to only give agents the context information they need to execute. All values
+/// accessed through context are read-only, and if modified, will not directly change the state of
+/// the simulation or any other agent.
```

Comment:
I actually can't find where this struct is used. @litvand will it be used in the Python runner, or is it just a remnant of the old cloud implementation. 

If it is used, is this a specific Agent's view into the context, or is it the context as a whole. I would guess it's a specific Agent's view as `neighbors` is a Vec instead of a map?

If so I suggest some changes along these lines:
```suggestion
/// The context is the global immutable part of state that is shared between all agents.
///
/// This struct is a specific Agent's view into the context. It contains information about the
/// general simulation, rather than data belonging to specific agents. This is effectively what the
/// agent 'can see'. Due to it being a description of the current environment surrounding the
/// agent, it's immutable (unlike an agent's specific state).

/// Language runners attempt to only give agents the context information they need to execute. 
/// All values accessed through context are read-only, and if modified, will not directly change
/// the state of the simulation or any other agent.
```